## Módulo 2 : Estadística y Visualización de Datos
#### Sesión : 01
#### Fecha : 01/08/20206
#### Autor : Jaime Gómez


In [6]:
print("Módulo 2 : Estadística y Visualización de Datos")

Módulo 2 : Estadística y Visualización de Datos


#### 1. Introducción a Numpy 

In [7]:
import numpy as np

# Creando un array a partir de una lista de Python
lista = [1, 2, 3, 4, 5]
# Creando un array de NumPy (ndarray) a partir de la lista
array = np.array(lista)

print(f"Array de Python: {lista} de tipo {type(lista)}")
print(f"Array de NumPy: {array} de tipo {type(array)}")


Array de Python: [1, 2, 3, 4, 5] de tipo <class 'list'>
Array de NumPy: [1 2 3 4 5] de tipo <class 'numpy.ndarray'>


##### **Ejercicio 1** 
Imprimir en un array de Numpy los nros de 2, 4, .., 200

In [8]:
import numpy as np
lista=[]
for i in range(2, 201, 2):
  lista.append(i)
array=np.array(lista)
print(array)

[  2   4   6   8  10  12  14  16  18  20  22  24  26  28  30  32  34  36
  38  40  42  44  46  48  50  52  54  56  58  60  62  64  66  68  70  72
  74  76  78  80  82  84  86  88  90  92  94  96  98 100 102 104 106 108
 110 112 114 116 118 120 122 124 126 128 130 132 134 136 138 140 142 144
 146 148 150 152 154 156 158 160 162 164 166 168 170 172 174 176 178 180
 182 184 186 188 190 192 194 196 198 200]


In [9]:
import numpy as np
numeros = np.arange(2, 201, 2)
print(numeros)

[  2   4   6   8  10  12  14  16  18  20  22  24  26  28  30  32  34  36
  38  40  42  44  46  48  50  52  54  56  58  60  62  64  66  68  70  72
  74  76  78  80  82  84  86  88  90  92  94  96  98 100 102 104 106 108
 110 112 114 116 118 120 122 124 126 128 130 132 134 136 138 140 142 144
 146 148 150 152 154 156 158 160 162 164 166 168 170 172 174 176 178 180
 182 184 186 188 190 192 194 196 198 200]


In [13]:
lista=[2*x for x in range(1,101)]
array=np.array(lista)
print(array)

[  2   4   6   8  10  12  14  16  18  20  22  24  26  28  30  32  34  36
  38  40  42  44  46  48  50  52  54  56  58  60  62  64  66  68  70  72
  74  76  78  80  82  84  86  88  90  92  94  96  98 100 102 104 106 108
 110 112 114 116 118 120 122 124 126 128 130 132 134 136 138 140 142 144
 146 148 150 152 154 156 158 160 162 164 166 168 170 172 174 176 178 180
 182 184 186 188 190 192 194 196 198 200]


In [14]:
# Creando un array a partir de una lista de Python
lista = [1, 2, 3, 4, 5]
# Creando un array de NumPy (ndarray) a partir de la lista
np_nros = np.array(lista)

## Otras propiedades de los arrays de NumPy
print("np_nros.dtype :", np_nros.dtype)
print("np_nros.ndim :", np_nros.ndim)
print("np_nros.size :", np_nros.size)
print("np_nros.shape :", np_nros.shape)
print("np_nros.itemsize :", np_nros.itemsize)
print("np_nros.data :", np_nros.data)

np_nros.dtype : int64
np_nros.ndim : 1
np_nros.size : 5
np_nros.shape : (5,)
np_nros.itemsize : 8
np_nros.data : <memory at 0x1068fc640>


2. Trabajando con arreglos

In [18]:
array_2d = [ [11, 15, 21],
               [33, 48 , 55],]

print(array_2d)

np_array_2d = np.array(array_2d)

print(np_array_2d)

[[11, 15, 21], [33, 48, 55]]
[[11 15 21]
 [33 48 55]]


In [19]:
print("np_array_2d.dtype :", np_array_2d.dtype)
print("np_array_2d.ndim :", np_array_2d.ndim)
print("np_array_2d.size :", np_array_2d.size)
print("np_array_2d.shape :", np_array_2d.shape)
print("np_array_2d.itemsize :", np_array_2d.itemsize)
print("np_array_2d.data :", np_array_2d.data)

np_array_2d.dtype : int64
np_array_2d.ndim : 2
np_array_2d.size : 6
np_array_2d.shape : (2, 3)
np_array_2d.itemsize : 8
np_array_2d.data : <memory at 0x106f182b0>


##### **Ejercicio 2**
Crea un arreglo de NumPy con 3 filas:

- Fila 1: los 10 primeros números pares.
- Fila 2: los 10 primeros números impares.
- Fila 3: secuencia 10, 30, 50, ..., 190.

In [20]:
fila_1 = np.arange(2, 21, 2)
fila_2 = np.arange(1, 20, 2)
fila_3 = np.arange(10, 101, 10)

array_3d = np.array([fila_1, fila_2, fila_3])

print(array_3d)

[[  2   4   6   8  10  12  14  16  18  20]
 [  1   3   5   7   9  11  13  15  17  19]
 [ 10  20  30  40  50  60  70  80  90 100]]


In [21]:
import numpy as np
fila1 = np.array([2*x for x in range(1, 11)])
fila2 = np.array([2*x - 1 for x in range(1, 11)])  
fila3 = np.array([10 + 20*x for x in range(0, 10)])
array=np.array([fila1, fila2, fila3])
print(array)

[[  2   4   6   8  10  12  14  16  18  20]
 [  1   3   5   7   9  11  13  15  17  19]
 [ 10  30  50  70  90 110 130 150 170 190]]


In [30]:
nros = [
    [ x for x in range(2,22,2)],
    [ y for y in range(1,21,2)],
    [ z for z in range(10,200,20)]
]
np_array = np.array(nros)
print(np_array)

[[  2   4   6   8  10  12  14  16  18  20]
 [  1   3   5   7   9  11  13  15  17  19]
 [ 10  30  50  70  90 110 130 150 170 190]]


In [31]:
import numpy as np

fila1 = []
fila2 = []
fila3 = []

# 10 primeros números pares
for x in range(2, 21, 2):
    fila1.append(x)

arrayf1=np.array(fila1)
print(np.array(arrayf1))



# 10 primeros números impares
for x in range(1, 20, 2):
    fila2.append(x)

arrayf2=np.array(fila2)
print(np.array(arrayf2))


# Secuencia 10, 30, 50, ..., 110
for x in range(10, 200, 20):
    fila3.append(x)

arrayf3=np.array(fila3)
print(np.array(arrayf3))

print(len(fila1), len(fila2), len(fila3))

arreglo = np.array([fila1, fila2, fila3])

print(arreglo)

[ 2  4  6  8 10 12 14 16 18 20]
[ 1  3  5  7  9 11 13 15 17 19]
[ 10  30  50  70  90 110 130 150 170 190]
10 10 10
[[  2   4   6   8  10  12  14  16  18  20]
 [  1   3   5   7   9  11  13  15  17  19]
 [ 10  30  50  70  90 110 130 150 170 190]]
